brave/brave-core
# Preliminary refactoring before adding system tray icon

Resolves https://github.com/brave/brave-browser/issues/31074

- Shared utilities between browser and wireguard service
- Updated service registration to install/uninstall/clean the service files from the service
- Changed service type start from auto to on demand
- Changed service executable type to be windowed instead of console
- Updated permissions for the config file to let administrators delete it



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Check that Vpn can be connected/disconnected using brave://flags/#brave-vpn-wireguard
- Check that after uninstalling the browser `c:\windows\temp\BraveVpn' folder removed
